### PR TITLE
[transform] fuse linalg.fill into gemm

### PR DESCRIPTION
**Pull request by wyzero**
to #787

### CODE REVIEW

File: tao_compiler/mlir/disc/tools/disc-transform/TransformOps/TransformOpsExt.cc [hunk 1]
```diff
@@ -191,6 +194,312 @@ DiagnosedSilenceableFailure DISCBufferizeOp::apply(
...
+      return failure();
+
+    rewriter.replaceOp(op, op.getSource());
+    return success();
+  }
+};
+
+/// convert:
+///   %0 = tensor.extract_slice %arg0[0, 0] [%s0, %s1] [1, 1] : tensor<?x?xf32>
+///   to tensor<?x?xf32> %1 = linalg.fill ins(%cst : f32) outs(%0 :
```
Comment (qiuxiafei) [inline tao_compiler/mlir/disc/tools/disc-transform/TransformOps/TransformOpsExt.cc:375]: nit: malformed...

File: tao_compiler/mlir/disc/tools/disc-transform/TransformOps/TransformOpsExt.cc [hunk 2]
```diff
@@ -191,6 +194,312 @@ DiagnosedSilenceableFailure DISCBufferizeOp::apply(
...
+    : public OpRewritePattern<tensor::ExtractSliceOp> {
+  using OpRewritePattern<tensor::ExtractSliceOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(tensor::ExtractSliceOp op,
+                                PatternRewriter& rewriter) const override {
+    auto producerOp = op.getSource().getDefiningOp<tensor::ExtractSliceOp>();
+    if (!strideAllOnes(op) || !producerOp || !strideAllOnes(producerOp))
+      return failure();
+
+    SmallVector<OpFoldResult> newOffsets;
```
[the remark this answers is not in the document]
Comment (wyzero) [reply tao_compiler/mlir/disc/tools/disc-transform/TransformOps/TransformOpsExt.cc:284]: yes, as you can see from the comment above.

File: tao_compiler/mlir/disc/tools/disc-transform/TransformOps/TransformOpsExt.cc (hunk 1 above)
Comment (wyzero) [reply tao_compiler/mlir/disc/tools/disc-transform/TransformOps/TransformOpsExt.cc:375]: ohhh, it's re-formatted by clang-format. I'll send a new PR today. To reduce CI overhead, I'll change it in that PR.

File: tao_compiler/mlir/disc/tools/disc-transform/TransformOps/TransformOpsExt.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (qiuxiafei) [reply tao_compiler/mlir/disc/tools/disc-transform/TransformOps/TransformOpsExt.cc:284]: oh ...  got it ..

**Review by qiuxiafei — APPROVED** [review]

**Pull request merged**